src/controllers/plugins/Plugin.hpp

```diff
@@ -154,6 +171,9 @@ class Plugin
     std::vector<QTimer *> activeTimeouts;
     int lastTimerId = 0;
 
+    OnDestroyed onDestroyed_;
```

Comment:
I think I'd prefer if the signals were publicly available, and consumers access them & call `connect` themselves